resilience4j-core/src/main/java/io/github/resilience4j/core/IntervalBiFunction.java

```diff
@@ -0,0 +1,18 @@
+package io.github.resilience4j.core;
+
+import io.vavr.control.Either;
+
+import java.util.function.BiFunction;
+
+/**
+ * An IntervalBiFunction which can be used to calculate the wait interval. The input parameters of the bi
+ * function is the number of attempts (attempt) and either result or exception, the output parameter is the wait interval in
+ * milliseconds. The attempt parameter starts at 1 and increases with every further attempt.
+ */
+@FunctionalInterface
+public interface IntervalBiFunction<T> extends BiFunction<Integer, Either<T, Throwable>, Long> {
```

Comment:
By convention, Either's Left represents a failure case and the Right represents a success.
T and `Throwable` must be switched.

Reply:
didn't know, will change 👍

Reply:
Sry, maybe it is even more clear, if we use `Try` instead of `Either`.

That means: `Either<T, Throwable>`  --> `Try<T>`
and `Either.right(...)` --> `Try.success(...)`
and `Either.left(...)` --> `Try.failure(...)`

and  `Throwable result = retryAttempt.isSuccess() ? null : retryAttempt.getCause();`